### ISSUE TITLE
Upload file to a specific folder in Canvas

### ISSUE THREAD

**Issue by elec3647** [ucfopen/canvasapi#189]
Hi,
Right now I can't seem to get a file named "Homework_11.pdf" uploaded to the "Homeworks/" folder under my course. The following example works:
```python
fh = open('path/to/Homework_11.pdf')
canvas = Canvas(API_URL, API_KEY)
course_id = 1113938
course = canvas.get_course(course_id)
course.upload(fh)
```
However, it gets uploaded to a new folder named "unfiled".

I sniffed the Canvas website to see what it does versus the canvasapi.upload.Uploader class.
It does the following:
`POST https://auburn.instructure.com/api/v1/folders/{folder_id}/files`

data submitted:
```
name: Homework_11.pdf
size: 231844
content_type: application/pdf
on_duplicate: rename
parent_folder_id: 5816809
no_redirect: true
_method: POST
authenticity_token:
```
So at this point it seems to follow the standard pattern of returning info on where to upload the file that the Uploader class handles. I am going to add a canvasapi.folder.Folder.upload() method to do this. Would be cool to pass a Folder object into class.upload(fh, folder=canvasapi.folder.Folder) to do this as well.

Let me know and I can send you guys what I got. Thank you so much for this library. It is a lifesaver!!!

**Comment by elec3647** [comment]
More info: https://canvas.instructure.com/doc/api/files.html#method.folders.create_file

**Comment by elec3647** [comment]
OK, here is my method:
```python
def upload(self, file, **kwargs):
        """
        Upload a file to this folder.

        :calls: `POST /api/v1/folders/:folder_id/files \
        <https://canvas.instructure.com/doc/api/files.html#method.folders.create_file>`_

        :param file: The file or path of the file to upload.
        :type file: file or str
        :returns: True if the file uploaded successfully, False otherwise, \
                    and the JSON response from the API.
        :rtype: tuple
        """
        my_path = 'folders/{}/files'.format(self.id)
        return Uploader(
            self._requester,
            my_path,
            file,
            **kwargs
        ).start() 
```

**Comment by elec3647** [comment]
OK, I think I got everything that's needed for the tests. Sorry for the delay! Please let me know how I can improve or what else needs to be corrected. :-)

**Comment by Thetwam** [comment]
Resolved in #213
